src/IO/HTTPCommon.h

```diff
@@ -58,6 +58,11 @@ using HTTPSessionPtr = std::shared_ptr<Poco::Net::HTTPClientSession>;
 /// All pooled sessions don't have this tag attached after being taken from a pool.
 /// If the request and the response were fully written/read, the client code should add this tag
 /// explicitly by calling `markSessionForReuse()`.
+///
+/// Note that HTTP response may contain extra bytes after the last byte of the payload. Specifically,
+/// when chunked encoding is used, there's an empty chunk at the end. Those extra bytes must also be
+/// read before the session can be reused. So we usually put an `istr->ignore(INT64_MAX)` call
```

Comment:
Shouldn't we put such comment near `istr->ignore(INT64_MAX)` call ? It's a different file and may be tricky to find it while reading `ReadBufferFromS3.cpp`